packages/app/src/components/DBTraceWaterfallChart.tsx

```diff
@@ -61,10 +58,45 @@ function getTableBody(tableModel: TSource) {
   }
 }
 
-function getConfig(source: TSource, traceId: string) {
+function useSortKey(source: TSource): string {
+  let rowSortTimestamp = source.timestampValueExpression;
+  const { data: sourceCols } = useColumns({
+    connectionId: source.connection,
+    ...source.from,
+  });
+  const defaultSelections = new Set(
+    source.defaultTableSelectExpression?.split(',').map(col => {
+      return col.trim();
+    }),
+  );
+  for (const col of sourceCols ?? []) {
+    if (defaultSelections.has(col.name) && col.type.startsWith('DateTime')) {
+      rowSortTimestamp = col.name;
+      break;
+    }
+  }
+  return source.kind === SourceKind.Trace
+    ? source.timestampValueExpression
+    : rowSortTimestamp;
```

Comment:
we want to keep `getDisplayedTimestampValueExpression(source)` as a fallback option